### PR TITLE
logging: File log handler changed to Stream handler. Sentry extension added.

### PR DESCRIPTION
**Pull request by oguzdemirbasci**
Ref: https://github.com/inspirehep/inspirehep/issues/1928

### CODE REVIEW

File: hepcrawl/extensions.py [hunk 1]
```diff
@@ -31,3 +34,21 @@ def spider_error(self, failure, response, spider, signal=None, sender=None, *arg
...
+    def from_crawler(cls, crawler):
+        sentry_dsn = crawler.settings.get('SENTRY_DSN', None)
+        if sentry_dsn is None:
+            raise NotConfigured
+        # instantiate the extension object
+        ext = cls()
+        # instantiate
+        sentry_sdk.init(sentry_dsn)
+        # return the extension object
+        return ext
```
Comment (MJedr) [inline hepcrawl/extensions.py:54]: missing new line

File: tests/functional/arxiv/test_arxiv.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def test_arxiv(
         crawler_instance=crawler,
         project=config['CRAWLER_PROJECT'],
         spider=spider,
-        settings={},
+        settings={'LOG_FILE':None},
```
Comment (MJedr) [inline tests/functional/arxiv/test_arxiv.py:110]:
```suggestion
        settings={'LOG_FILE': None},
```

File: tests/unit/test_extensions.py [hunk 3]
```diff
@@ -40,3 +42,21 @@ def test_error_handler(crawler):
...
+
+@mock.patch("hepcrawl.extensions.sentry_sdk.init")
+def test_sentry_logging_init(mock_sentry_sdk, crawler):
+    """Test SentryLogging extension."""
+    log_settings = {
+        "SENTRY_DSN" : "TEST_SENTRY_DSN",
+        "EXTENSIONS" : {
+            'hepcrawl.extensions.SentryLogging': 100,
+            'scrapy_sentry.extensions.Errors': 200,
+            'hepcrawl.extensions.ErrorHandler': 300,
```
Comment (MJedr) [inline tests/unit/test_extensions.py:55]: It's not already in project settings? Or are settings for tests different?

**Review by MJedr — COMMENTED** [review]
Looks good for me, some minor comments. Good job 👏

File: tests/unit/test_extensions.py (hunk 3 above)
Comment (oguzdemirbasci) [reply tests/unit/test_extensions.py:55]: In `settings.py`, it is set only if environment variable `APP_SENTRY_DSN` exists which defined in kubernetes. So I had to define them seperately in tests.

**Review by MJedr — APPROVED** [review]

**Pull request merged**